website/docs/r/storage_account_local_user.html.markdown

```diff
@@ -0,0 +1,142 @@
+
+resource "azurerm_storage_account_local_user" "example" {
+  name                 = "user1"
+  storage_account_name = azurerm_storage_account.example.name
+  ssh_key_enabled      = true
+  ssh_password_enabled = true
+  home_directory       = "example_path"
+  ssh_authorized_key {
+    description = "key1"
+    key         = local.second_public_key
+  }
+  ssh_authorized_key {
+    description = "key2"
+    key         = local.second_public_key
+  }
+  permission_scope {
+    permissions {
+      read   = true
+      create = true
+    }
+    service       = "blob"
+    resource_name = azurerm_storage_container.example.name
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Storage Account Local User. Changing this forces a new Storage Account Local User to be created.
+
+* `storage_account_name` - (Required) The name of the Storage Account that this Storage Account Local User resides in. Changing this forces a new Storage Account Local User to be created.
+
+---
+
+* `home_directory` - (Optional) The home directory of the Storage Account Local User.
+
+* `permission_scope` - (Optional) One or more `permission_scope` blocks as defined below.
+
+* `ssh_authorized_key` - (Optional) One or more `ssh_authorized_key` blocks as defined below. Changing this forces a new Storage Account Local User to be created.
+
+* `ssh_key_enabled` - (Optional) Should the SSH key be enabled?
+
+* `ssh_password_enabled` - (Optional) Should the SSh password be enabled?
+
+---
+
+A `permission_scope` block supports the following:
+
+* `permissions` - (Required) A `permissions` block as defined below.
+
+* `resource_name` - (Required) The container name (when `service` is set to `blob`) or the file share name (when `service` is set to `file`), used by the Storage Account Local User.
+
+* `service` - (Required) The storage service used by this Storage Account Local User. Possible values are `blob` and `file`.
+
+---
+
+A `permissions` block supports the following:
+
+* `create` - (Optional) The permission to create the resources defined in this scope.
+
+* `delete` - (Optional) The permission to delete the resources defined in this scope.
+
+* `list` - (Optional) The permission to list the resources defined in this scope.
+
+* `read` - (Optional) The permission to read the resources defined in this scope.
+
+* `write` - (Optional) The permission to modify the resources defined in this scope.
+
+---
+
+A `ssh_authorized_key` block supports the following:
+
+* `description` - (Required) The description of this SSH authorized key.
+
+* `key` - (Required) The public key value of this SSH authorized key.
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the Storage Account Local User.
+
+* `password` - The value of the password, which is only available when `ssh_password_enabled` is set to `true`.
```

Comment:
we should call out if this gets cycled every time during update or not too, since users may need to account for this - and we may also need to `SetNewComputed` in a customizeDiff?